spec/childprocess_spec.rb

```diff
@@ -107,6 +107,23 @@
       expect(child_env['CHILD_ONLY']).to eql '1'
     end
   end
+  
+  it 'allows unicode characters in the environment' do
+    # This test does not work on Windows for Ruby < 2.3 because ENV will not be properly decoded
+    # This was fixed in Ruby 2.3 here: https://github.com/ruby/ruby/commit/5e3467c4414df815b3b00d2b0372026b069e7f7d
```

Comment:
Not that ruby only fixed using UTF8 characters in ENV _VALUES_ not keys

If you change the test the code to
``` ruby
    Tempfile.open("env-spec") do |file|
      process = write_env(file.path)
      process.environment['baör'] = 'baör'
      process.start
      process.wait
      
      child_env = eval rewind_and_read(file)
      
      expect(child_env['baör'].force_encoding('UTF-8')).to eql 'baör'
    end
```

Inspecting `child_env` in pry you can see the corruption
```
[1] pry(#<RSpec::ExampleGroups::ChildProcess>)> child_env
=> {"ALLUSERSPROFILE"=>"C:\\ProgramData",
 "ANSICON"=>"185x2000 (185x65)",
 "ANSICON_DEF"=>"7",
 "APPDATA"=>"C:\\Users\\glenn.sarti\\AppData\\Roaming",
 "BUNDLER_VERSION"=>"1.15.4",
 "BUNDLE_BIN_PATH"=>"C:/tools/ruby2.3.1x64/lib/ruby/gems/2.3.0/gems/bundler-1.15.4/exe/bundle",
 "BUNDLE_GEMFILE"=>"C:/Source/tmp/childprocess/Gemfile",
 "ChocolateyInstall"=>"C:\\ProgramData\\chocolatey",
 "ChocolateyLastPathUpdate"=>"Sun Aug 27 19:31:25 2017",
 "ChocolateyToolsLocation"=>"C:\\tools",
...
"ba\xC3\xB6r"=>"ba\xC3\xB6r"}
```

Also note that this testing method mangles any existing UTF8 env vars
e.g. in PowerShell create a UTF8 envvar
``` powershell
PS> $ENV:UTF8ᚠᛇᚻ = 'UTF8ᚠᛇᚻ'
PS> dir ENV:

Name                           Value
----                           -----
ALLUSERSPROFILE                C:\ProgramData
ANSICON                        100x2000 (100x33)
ANSICON_DEF                    7
APPDATA                        C:\Users\glenn.sarti\AppData\Roaming
ChocolateyInstall              C:\ProgramData\chocolatey
...
UTF8ᚠᛇᚻ                        UTF8ᚠᛇᚻ
```

Then run the spec test and pry into `child_env`
```
...
 "USERNAME"=>"glenn.sarti",
 "USERPROFILE"=>"C:\\Users\\glenn.sarti",
 "UTF8???"=>"UTF8???",
 "windir"=>"C:\\WINDOWS",
 "ba\xC3\xB6r"=>"ba\xC3\xB6r"}
```

Note the UTF8 characters are now `?` and the UTF8 from childprocess is now bytes.  This has not been fixed in Ruby 2.4.x yet either.